File: dotNet/NetworkResult/NetworkResult/Models/TrainStats.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace NetworkResult.Models
{
    class TrainStats
    {
        [JsonPropertyName("accuracy")]
        public float Accuracy { get; set; }
        
        [JsonPropertyName("k1")]
        public float K1 { get; set; }
        
        [JsonPropertyName("k5")]
        public float K5 { get; set; }
        
        [JsonPropertyName("model_name")]
        public string ModelName { get; set; }
        
        [JsonPropertyName("train_loss_array")]
        public List<float> TrainLoss { get; set; }
        
        [JsonPropertyName("valid_loss_array")]
        public List<float> ValidLoss{ get; set; }
        
        [JsonPropertyName("train_accuracy_array")]
        public List<float> TrainAccuracy { get; set; }
        
        [JsonPropertyName("valid_accuracy_array")]
        public List<float> ValidAccuracy{ get; set; }
        
        [JsonPropertyName("best_epoch")]
        public int BestEpoch { get; set; }
        
        [JsonPropertyName("total_time")]
        public float TotalTime{ get; set; }
        
        [JsonPropertyName("train_time_sum")]
        public float TrainTime{ get; set; }
        
        [JsonPropertyName("eval_time_sum")]
        public float EvalTime{ get; set; }
    }
}

File: trainstats.py
from pathlib import Path
from datetime import datetime
from datasetmodel import DatasetModel
import numpy as np
import json


class TrainStats:
    accuracy = 0
    k1 = 0
    k5 = 0
    class_accuracy = {}
    confusion = np.ndarray(shape=(2, 2))

    def __init__(self, model_name, train_loss_array, valid_loss_array, train_accuracy_array, valid_accuracy_array, best_epoch, total_time, train_time_sum, eval_time_sum):
        self.model_name = model_name
        self.train_loss_array = train_loss_array
        self.valid_loss_array = valid_loss_array
        self.train_accuracy_array = train_accuracy_array
        self.valid_accuracy_array = valid_accuracy_array
        self.best_epoch = best_epoch
        self.total_time = total_time
        self.train_time_sum = train_time_sum
        self.eval_time_sum = eval_time_sum

    def save(self, working_ds: DatasetModel):
        now = datetime.now().strftime("%Y-%m-%d %H-%M-%S")
        dir_path = f'drive/MyDrive/results/{working_ds.name}_results/'
        file_name = f'{self.model_name}_{now}'

        Path(dir_path).mkdir(parents=True, exist_ok=True)

        with open(f'{dir_path}{file_name}.json', 'w') as outfile:
            json.dump(self.__dict__, outfile)
        print(f'💾File {file_name}.json saved at {dir_path}')
